tests/sles4sap/hana_install.pm

```diff
@@ -41,18 +43,28 @@ sub run {
     # Add host's IP to /etc/hosts
     $self->add_hostname_to_hosts;
 
+    # Install libopenssl1_0_0 for older (<SPS03) HANA versions on SLE15+
+    my $hana_version = $path =~ s/.*\/SPS([0-9]+)rev[0-9]\/.*/$1/r;
```

Comment:
In case this regexp fails, the content of `$hana_version` will be the same as the content of `$path`, and probably that string will be converted to 0 if evaluated as a number. Perhaps you can add a verification for that case here, or check that the regexp worked. For example:

```
if ($path =~ /.*\/SPS([0-9]+)rev[0-9]\/.*/$1/) {
    my $hana_version = $1;
    if (is_sle('15+') && ($hana_version <= 2)) {
        # The old libopenssl is in Legacy Module
        add_suseconnect_product('sle-module-legacy');
        zypper_call('in libopenssl1_0_0');
    }
}
```
It's not so dangerous because worst case, Legacy module is added and libopenssl1 is installed, but it may not be needed.

Reply:
Thanks, I added this test.